### ISSUE TITLE
Getting Command Name in Before Action

### ISSUE THREAD

**Issue by janwiemers** [urfave/cli#635]
Hi there,

I just started with Go and this framework.
I'm trying to unify some code right now and would like to outsource a check I do in every sub-command except one.
I was thinking that I potentially can use the `app.Before` hook to and skip the check for the exception case. Unfortunatelly I haven't found a way to get the current command name.
Is the name available to the Before func?

Thanks
Jan

**Comment by jszwedko** [comment]
Hi @janwiemers,

The subcommand is determined after the `Before` function is called on the `App` (https://github.com/urfave/cli/blob/master/app.go#L234-L243) so unfortunately the command that would be executed (if any) is not available in the `Before` function.

**Comment by phinnaeus** [comment]
How about the After function? I see the way the code is written currently that the After function call is deferred right before Before is called and therefore still prior to the Command being resolved, but this could be changed. Intuitively I would expect the After function context to have the same contents as the Command Action context.

**Comment by jszwedko** [comment]
@phinnaeus the command name is available on the `cli.Context` passed into the `After` function. The call is `defer`ed, but the `context` variable is bound to the outer scope and so changes to it (including populating the `.Command`) are available.

**Comment by jszwedko** [comment]
Example:

```go
package main                                                                                                                                          

import (
        "fmt"
        "os"

        "github.com/urfave/cli"
)

func main() {
        app := cli.NewApp()

        app.Commands = []cli.Command{
                {
                        Name:    "add",
                        Aliases: []string{"a"},
                        Usage:   "add a task to the list",
                        Action: func(c *cli.Context) error {
                                return nil
                        },
                        After: func(c *cli.Context) error {
                                fmt.Printf("command: %s\n", c.Command.Name)
                                return nil
                        },
                },
        }

        app.Run(os.Args)
}
```

```sh
➜  /tmp go run /tmp/tmp.go add                                                                                                                        
command: add  
```

**Comment by phinnaeus** [comment]
My bad, I tested with an older version. Thanks for clarifying.

**Comment by lynncyrin** [comment]
Given that this is from last year, I think I'm comfortable closing it 🙂 feel free to re-open / open a new issue / comment in support if there's still interest here!

**Comment by gurleensethi** [comment]
You can use the `Args` function, in the cli context, to figure out the command in the `Before` function.

```go
function Before(cliCtx *cli.Context) error {
	if cliCtx.NArg() >= 1 && cliCtx.Args().First() == "your command" {
				return nil
	}
}
```

This worked for me, you use case may vary.
